sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1719,3 +1720,20 @@ object OptimizeLimitZero extends Rule[LogicalPlan] {
       empty(ll)
   }
 }
+
+/**
+ * If the UDF is deterministic and if the children are all literal, we can replace the udf
+ * with the output of the udf serialized
+ */
+object DeterministicLiteralUDF extends Rule[LogicalPlan] {
```

Comment:
Could you make a separate file for this new optimizer rule?

Reply:
Thanks @dongjoon-hyun for the comment.  Sure will take a look.

Reply:
Moved rule to a new file in the last push.